secret/secrets_manager_vault.go

```diff
@@ -20,20 +22,36 @@ func NewBasicSecretsManager(c SecretsManagerClient) *BasicSecretsManager {
 
 // CreateSecret creates a new secret.
 func (m *BasicSecretsManager) CreateSecret(ctx context.Context, s NamedSecret) (id string, err error) {
-	return "", errors.New("TODO: implement")
+	out, err := m.client.CreateSecret(ctx, &secretsmanager.CreateSecretInput{
+		Name:         s.Name,
+		SecretString: s.Value,
+	})
+	return *out.ARN, err
 }
 
 // GetValue returns an existing secret's decrypted value.
 func (m *BasicSecretsManager) GetValue(ctx context.Context, id string) (val string, err error) {
-	return "", errors.New("TODO: implement")
+	out, err := m.client.GetSecretValue(ctx, &secretsmanager.GetSecretValueInput{SecretId: &id})
```

Comment:
Check for empty ID before calling `GetSecretValue`. While Secrets Manager will likely fail anyways, it's better to check inputs before making network calls because network calls are expensive, so if you know in advance that an input is nonsense, validate it in advance of making the expensive call.